python/graphstorm/model/hgt_encoder.py

```diff
@@ -0,0 +1,288 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Heterogeneous Graph Transformer (HGT) layer implementation
+"""
+import warnings
+
+import math
+import torch
+import torch.nn.functional as F
+import dgl.function as fn
+
+from torch import nn
+from dgl.nn.functional import edge_softmax
+from .ngnn_mlp import NGNNMLP
+from .gnn_encoder_base import GraphConvEncoder
+
+
+class HGTLayer(nn.Module):
+    r"""Heterogenous graph transformer (HGT) layer.
+    
+    Different from DGL's HGTConv, this implementation is based on heterogeneous graph.
+    Other hyperparameters' default values are same as the DGL's HGTConv setting.
+    
```

Comment:
Can you give the formula of HGTConv here?

Reply:
add to the encoder class. This will help to automatically build API docs.